lib/tools.js

```diff
@@ -82,26 +83,41 @@ function asJson (obj, msg, num, time) {
   // we need the child bindings added to the output first so instance logged
   // objects can take precedence when JSON.parse-ing the resulting log line
   data = data + chindings
+
+  if (hasObj === false) return data + end
+
+  const notHasOwnProperty = obj.hasOwnProperty === undefined
+
+  if (objError === true) {
+    data += ',"type":"Error","stack":' + stringify(obj.stack)
+  }
+
+  // if global serializer is set, call it first
+  if (serializers[wildcardGsym]) {
+    obj = serializers[wildcardGsym](obj)
+  }
+
+  var hasSerializableKeys = false
+  for (var szKey in serializers) {
+    if (szKey in obj) hasSerializableKeys = true
+  }
```

Comment:
Instead of iterating over serializers, can you compile a function with all the set serializers names?

Reply:
can try it

Reply:
if we do that it will down the creation of slow child loggers with serializers  - they'll have to compile a funciton. that's going to be pretty expensive...

Reply:
of course, doing that will negatively impact http logging

Reply:
There is no need to do that for every child instantiation. The vast majority of the time child loggers do not have serializers, so we can reuse the function that was already there.